### PR TITLE
Update yard

### LINKED ISSUE #24: Host - Edit Yard
As a host,
When I visit the yard show page,
I want to click the button to edit,
So I can be taken to the edit yard form where I will see the yard information including:

- [x] name
- [x] location
- [x] description
- [x] availability
- [x] photo URLs
- [x]  Checkboxes next to booking purpose

When I fill out the form and click ‘Update Yard’
Then I am taken back to the yard show page
And I see the information updated.

Wireframe:
Screen Shot 2021-04-12 at 7 12 06 PM

### LINKED ISSUE #28: Host - Edit Yard Sad Path
As a host, 
When I visit the edit yard page, and I don't enter either a 
- [x] name
- [x] location
- [x] OR check at least one booking purpose

A flash message notifies me that the I cannot update the yard with missing information.

### PR DESCRIPTION
**Pull request by amsmyth1**
**What Changed**
- Add view and HostYard Controller method for update
- Add testing for happy path
- Add testing for sad path.

close #24 #28 

**Known Issues**
- Purposes do not pre-populate with data

**What is Next**
- Refactor to utilize partial for form
- Refactor into facade (cannot be same facade as YardsFacade as address needs to be stored differently)

### CODE REVIEW

File: app/controllers/host/yards_controller.rb [hunk 1]
```diff
@@ -17,11 +17,67 @@ def create
     end
   end
 
+  def edit
+    @yard = yard_details(params[:id])
```
Comment (jbranham1) [inline app/controllers/host/yards_controller.rb]: Can we use the YardFacade for this?

File: app/controllers/host/yards_controller.rb [hunk 2]
```diff
@@ -17,11 +17,67 @@ def create
...
+                                photo_url_2:    yard[:attributes][:photo_url_2],
+                                photo_url_3:    yard[:attributes][:photo_url_3] })
+    end
+  end
+  
+  def all_yard_purposes(yard)
+    yard[:attributes][:purposes][:data].map do |purpose|
+      purpose
+    end
+  end
```
Comment (jbranham1) [inline app/controllers/host/yards_controller.rb]: Same as above! This is all in the yard facade.

**Review by jbranham1 — COMMENTED** [review]
I left a few comments!

File: app/controllers/host/yards_controller.rb [hunk 3]
```diff
@@ -17,11 +17,67 @@ def create
     end
   end
 
+  def edit
+    @yard = yard_details(params[:id])
+  end
+
+  def update
+    params[:_method] == "patch"
```
Comment (Gvieve) [inline app/controllers/host/yards_controller.rb]: I think we can add this to the edit form rather than having it in the controller.

**Review by Gvieve — APPROVED** [review]
Overall looks great. Thanks for tackling this. I left one small comment and second what Jenny mentioned.

File: app/controllers/host/yards_controller.rb (hunk 1 above)
Comment (amsmyth1) [reply app/controllers/host/yards_controller.rb]: We can't it is slightly different with the street address, YardFacade combines the fields for display, the edit method needs the fields individual for the form info to prepopulate

**Review by jbranham1 — APPROVED** [review]

File: app/controllers/host/yards_controller.rb (hunk 3 above)
Comment (amsmyth1) [reply app/controllers/host/yards_controller.rb]: I was able to get rid of this all together, great catch!

**Pull request merged**